util/util.go

```diff
@@ -410,3 +412,39 @@ func IsPaused(cluster *clusterv1.Cluster, v metav1.Object) bool {
 	_, ok := annotations[clusterv1.PausedAnnotation]
 	return ok
 }
+
+func GetCRDWithContract(ctx context.Context, c client.Client, gvk schema.GroupVersionKind, contract string) (*apiextensionsv1.CustomResourceDefinition, error) {
+	crdList := &apiextensionsv1.CustomResourceDefinitionList{}
+	for {
+		if err := c.List(ctx, crdList, client.Continue(crdList.Continue), client.HasLabels{contract}); err != nil {
+			return nil, errors.Wrapf(err, "failed to list CustomResourceDefinitions for %v", gvk)
+		}
+
+		for _, crd := range crdList.Items {
+			if crd.Spec.Group == gvk.Group &&
+				crd.Spec.Names.Kind == gvk.Kind {
+				return crd.DeepCopy(), nil
+			}
+		}
+
+		if crdList.Continue == "" {
+			break
+		}
+	}
+
+	return nil, errors.Errorf("failed to find a CustomResourceDefinition for %v with contract %q", gvk, contract)
+}
+
+// KubeVersions is a sortable slice of kube-like version strings.
+//
+// Kube-like version strings are starting with a v, followed by a major version,
+// optional "alpha" or "beta" strings followed by a minor version (e.g. v1, v2beta1).
+// Versions will be sorted based on GA/alpha/beta first and then major and minor
+// versions. e.g. v2, v1, v1beta2, v1beta1, v1alpha1.
+type KubeVersions []string
```

Comment:
KubeAwareVersion?

Reply:
+1, `KubeVersions` instantly makes me think of KubernetesVersion.

Reply:
Opted for `KubeAwareAPIVersions` to be extra clear :)